Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,491 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+             set rᵣ : StructuredArrow WalkingReflexivePair.one inclusionWalkingReflexivePair :=
+                StructuredArrow.mk (Y := one) right
+             have hrₗ : rₗ ∈ p := by
+              · suffices f : e₀ ⟶  rₗ by exact (h₂ f).mp h₁
+                exact StructuredArrow.homMk .left
+             have hrᵣ : rᵣ ∈ p := by
+              · suffices f : e₀ ⟶  rᵣ by exact (h₂ f).mp h₁
+                exact StructuredArrow.homMk .right
+             cases y <;> cases f
+             rotate_right 3
+             · exact h₁
+             · exact hrₗ
+             · exact hrᵣ
+             · set v : StructuredArrow .one inclusionWalkingReflexivePair :=
+                StructuredArrow.mk (Y := zero) leftCompReflexion
+               suffices f : v ⟶  rₗ by exact (h₂ f).mpr hrₗ
+               exact StructuredArrow.homMk .left
+             · set v : StructuredArrow .one inclusionWalkingReflexivePair :=
+                StructuredArrow.mk (Y := .zero) rightCompReflexion
+               suffices f : v ⟶  rᵣ by exact (h₂ f).mpr hrᵣ
+               exact StructuredArrow.homMk .right
+
+end WalkingParallelPair
+
+end Limits
+
+namespace Limits
+
+open WalkingReflexivePair
+
+variable {C : Type u} [Category.{v} C]
+
+variable {A B : C}
+
+/-- Bundle the data of a parallel pair along with a common section as a functor out of the walking
+reflexive pair -/
+def reflexivePair (f g : A ⟶ B) (s : B ⟶ A)
+    (sl : s ≫ f = 𝟙 B := by aesop_cat) (sr : s ≫ g = 𝟙 B := by aesop_cat) :
+    (WalkingReflexivePair ⥤ C) where
+  obj x :=
+    match x with
+    | zero => B
+    | one => A
+  map h :=
+    match h with
+    | .id _ => 𝟙 _
+    | .left => f
+    | .right => g
+    | .reflexion => s
+    | .rightCompReflexion => g ≫ s
+    | .leftCompReflexion => f ≫ s
+  map_comp := by
+    rintro _ _ _ ⟨⟩ g <;> cases g <;>
+      simp only [Category.id_comp, Category.comp_id, Category.assoc, sl, sr,
+        reassoc_of% sl, reassoc_of% sr] <;> rfl
+
+
+/-- (Noncomputably) bundle the data of a reflexive pair as a functor out of the walking reflexive
+pair -/
+noncomputable def ofIsReflexivePair (f g : A ⟶ B) [IsReflexivePair f g] :
+    WalkingReflexivePair ⥤ C := reflexivePair f g (commonSection f g)
+
+/-- The natural isomorphism between the diagram obtained by forgetting the reflexion of
+`ofIsReflexivePair f g` and the original parallel pair. -/
+noncomputable def inclusionWalkingReflexivePairOfIsReflexivePairIso
+    (f g : A ⟶ B) [IsReflexivePair f g] :
+    WalkingParallelPair.inclusionWalkingReflexivePair ⋙ (ofIsReflexivePair f g) ≅
+      parallelPair f g :=
+  diagramIsoParallelPair _
+
+end Limits
+
+namespace Limits
+
+variable {C : Type u} [Category.{v} C]
+
+namespace reflexivePair
+
+open WalkingReflexivePair WalkingReflexivePair.Hom
+
+section
+
+variable {A B : C}
+variable (f g : A ⟶ B) (s : B ⟶ A) (sl : s ≫ f = 𝟙 B) (sr : s ≫ g = 𝟙 B)
+
+@[simp] lemma obj_zero : (reflexivePair f g s sl sr).obj zero = B := rfl
```

Comment:
These lemmas should preferably be stated just after the definition of `reflexivePair`. The standard names should be `reflexivePair_obj_zero`, etc.